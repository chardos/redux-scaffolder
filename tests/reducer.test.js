const rsk = require('../src/index.js');
const fs = require('fs-extra');
const makeDir = require('make-dir');
const del = require('del');
const tmpPath = `${process.cwd()}/tmp`;

describe('"reducer" command', () => {
  beforeEach(async () => {
    await makeDir('tmp');
    process.argv.splice(2);
  });

  afterEach(async () => {
    await del(`${tmpPath}/*`);
  })

  describe('When creating a new reducer', () => {
    describe('when style = "rails"', () => {
      it('the files should match snapshot', async () => {
        process.argv.push('setup-store', '--codeDirectory=tmp');
        await rsk();
        process.argv.splice(2);
        process.argv.push('reducer', 'sports', 'volleyball', 'soccer', '--codeDirectory=tmp');
        await rsk();
        const actionOutput = await fs.readFile(`${tmpPath}/store/sports/actions.js`);
        const reducerOutput = await fs.readFile(`${tmpPath}/store/sports/reducer.js`);
        
        expect([
          actionOutput.toString(),
          reducerOutput.toString()
        ]).toMatchSnapshot()
      })
    });
  })
})