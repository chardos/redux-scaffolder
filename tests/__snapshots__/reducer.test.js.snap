// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"reducer" command When creating a new reducer when combineActionsAndReducers is false the files should match snapshot 1`] = `
Array [
  "export const VOLLEYBALL = \\"VOLLEYBALL\\";
export const SOCCER = \\"SOCCER\\";

export const volleyball = () => {
  return {
    type: VOLLEYBALL
  };
};

export const soccer = () => {
  return {
    type: SOCCER
  };
};
",
  "import { VOLLEYBALL, SOCCER } from \\"../actions/sports\\";

export default function sports(state = {}, action) {
  switch (action.type) {
    case VOLLEYBALL:
      return state;

    case SOCCER:
      return state;

    default:
      return state;
  }
}
",
]
`;
